device/Microsoft.Azure.Devices.Client/Transport/AmqpTransportHandler.cs

```diff
@@ -674,8 +697,7 @@ private Twin TwinFromResponse(AmqpMessage message)
             {
                 string body = reader.ReadToEnd();
                 var props = JsonConvert.DeserializeObject<Microsoft.Azure.Devices.Shared.TwinProperties>(body);
-                var twin = new Twin();
-                twin.Properties = props;
+                var twin = new Twin(props);
```

Comment:
Nit: `return new Twin(props);` would be simpler.

Reply:
Fixed